app/src/main/java/com/duckduckgo/app/browser/omnibar/OmnibarLayout.kt

```diff
@@ -603,6 +608,10 @@ open class OmnibarLayout @JvmOverloads constructor(
         }
     }
 
+    private fun shouldShowUpdatedPrivacyShield(navigationBarEnabled: Boolean): Boolean {
+        return senseOfProtectionExperiment.shouldShowNewPrivacyShield() || navigationBarEnabled
+    }
```

Comment:
I'm not a fan of logic like this baked into the view (mainly for clarity and  ease of testing reasons) - we make one if-check here and then another in `BrowserLottieTrackersAnimatorHelper.kt` to finally select the animation.

What do you think about turning `OmnibarLayoutViewModel.LeadingIconState` into a sealed class? That way, the view model could provide something like `OmnibarLayoutViewModel.LeadingIconState.PRIVACY_SHIELD.resourceType`, where `resourceType` defines which animation to use (out of production, sense of protection, or the new visual update).

I'm not going to block the PR over this, but if you agree this is a worthwhile improvement, it’d be great to incorporate here or as a follow up.

Reply:
I don’t like this logic either, and ideally we’d refactor `BrowserLottieTrackersAnimatorHelper.kt` but that’s not something I want to do now. 

The problem is that now we have 3 different tracker animations and 2 of them have different layouts. This is far from optimal, but it solves the issue now knowing that this will be cleaned up once the experiments finish.